### ISSUE TITLE
Python `__init__.py` file missing?

### ISSUE THREAD

**Issue by VolkerH** [AdaptiveParticles/LibAPR#110]
Hi,

I'm trying to get the python samples to work, but I am wondering whether some python code is missing from the repository. Specifically I assume I would need something like an `__init.py__` to load pyApr as a module. 

I built  libapr with cmake using the option `-DAPR_BUILD_PYTHON_WRAPPERS=ON`. I'm not sure whether the cmake step should produce the `.py` files that I need to import pyAPR. 
I can see that I have a `pyApr.so` in the build folder and I'm adding that folder to the system path.
However,  I can't `import pyApr`.

For reference: I've been building this inside the following docker environment:
https://github.com/VolkerH/MyDockerStuff/tree/master/my_libapr

the build folder looks as follows:

```
root@9d00051e6719:/srv/LibAPR/build# tree
.
|-- CMakeCache.txt
|-- CMakeFiles
|   |-- 3.10.2
|   |   |-- CMakeCCompiler.cmake
|   |   |-- CMakeCXXCompiler.cmake
|   |   |-- CMakeDetermineCompilerABI_C.bin
|   |   |-- CMakeDetermineCompilerABI_CXX.bin
|   |   |-- CMakeSystem.cmake
|   |   |-- CompilerIdC
|   |   |   |-- CMakeCCompilerId.c
|   |   |   |-- a.out
|   |   |   `-- tmp
|   |   `-- CompilerIdCXX
|   |       |-- CMakeCXXCompilerId.cpp
|   |       |-- a.out
|   |       `-- tmp
|   |-- CMakeDirectoryInformation.cmake
|   |-- CMakeOutput.log
|   |-- CMakeTmp
|   |-- FindOpenMP
|   |   |-- OpenMPCheckVersion.c
|   |   |-- OpenMPCheckVersion.cpp
|   |   |-- OpenMPTryFlag.c
|   |   |-- OpenMPTryFlag.cpp
|   |   |-- ompver_C.bin
|   |   `-- ompver_CXX.bin
|   |-- Makefile.cmake
|   |-- Makefile2
|   |-- TargetDirectories.txt
|   |-- aprObjLib.dir
|   |   |-- C.includecache
|   |   |-- CXX.includecache
|   |   |-- DependInfo.cmake
|   |   |-- build.make
|   |   |-- cmake_clean.cmake
|   |   |-- depend.internal
|   |   |-- depend.make
|   |   |-- flags.make
|   |   |-- progress.make
|   |   `-- src
|   |       `-- io
|   |           |-- blosc_filter.c.o
|   |           `-- hdf5functions_blosc.cpp.o
|   |-- cmake.check_cache
|   |-- feature_tests.bin
|   |-- feature_tests.c
|   |-- feature_tests.cxx
|   |-- hdf5
|   |   |-- CMakeFiles
|   |   |   `-- CMakeTmp
|   |   `-- cmake_hdf5_test.c
|   |-- progress.marks
|   |-- pyApr.dir
|   |   |-- CXX.includecache
|   |   |-- DependInfo.cmake
|   |   |-- build.make
|   |   |-- cmake_clean.cmake
|   |   |-- depend.internal
|   |   |-- depend.make
|   |   |-- flags.make
|   |   |-- link.txt
|   |   |-- progress.make
|   |   `-- src
|   |       `-- wrapper
|   |           `-- pythonBind.cpp.o
|   `-- sharedLib.dir
|       |-- DependInfo.cmake
|       |-- build.make
|       |-- cmake_clean.cmake
|       |-- depend.internal
|       |-- depend.make
|       |-- flags.make
|       |-- link.txt
|       `-- progress.make
|-- ConfigAPR.h
|-- Makefile
|-- cmake_install.cmake
|-- external
|   |-- c-blosc
|   |   |-- CMakeFiles
|   |   |   |-- CMakeDirectoryInformation.cmake
|   |   |   `-- progress.marks
|   |   |-- Makefile
|   |   |-- blosc
|   |   |   |-- CMakeFiles
|   |   |   |   |-- CMakeDirectoryInformation.cmake
|   |   |   |   |-- blosc_static.dir
|   |   |   |   |   |-- C.includecache
|   |   |   |   |   |-- CXX.includecache
|   |   |   |   |   |-- DependInfo.cmake
|   |   |   |   |   |-- __
|   |   |   |   |   |   `-- internal-complibs
|   |   |   |   |   |       |-- lizard-1.0
|   |   |   |   |   |       |   |-- lizard_compress.c.o
|   |   |   |   |   |       |   |-- lizard_decompress.c.o
|   |   |   |   |   |       |   `-- lizard_frame.c.o
|   |   |   |   |   |       |-- lz4-1.7.5
|   |   |   |   |   |       |   |-- lz4.c.o
|   |   |   |   |   |       |   `-- lz4hc.c.o
|   |   |   |   |   |       |-- snappy-1.1.1
|   |   |   |   |   |       |   |-- snappy-c.cc.o
|   |   |   |   |   |       |   |-- snappy-sinksource.cc.o
|   |   |   |   |   |       |   |-- snappy-stubs-internal.cc.o
|   |   |   |   |   |       |   `-- snappy.cc.o
|   |   |   |   |   |       `-- zstd-1.3.0
|   |   |   |   |   |           |-- common
|   |   |   |   |   |           |   |-- entropy_common.c.o
|   |   |   |   |   |           |   |-- error_private.c.o
|   |   |   |   |   |           |   |-- fse_decompress.c.o
|   |   |   |   |   |           |   |-- pool.c.o
|   |   |   |   |   |           |   |-- threading.c.o
|   |   |   |   |   |           |   |-- xxhash.c.o
|   |   |   |   |   |           |   `-- zstd_common.c.o
|   |   |   |   |   |           |-- compress
|   |   |   |   |   |           |   |-- fse_compress.c.o
|   |   |   |   |   |           |   |-- huf_compress.c.o
|   |   |   |   |   |           |   |-- zstd_compress.c.o
|   |   |   |   |   |           |   `-- zstdmt_compress.c.o
|   |   |   |   |   |           `-- decompress
|   |   |   |   |   |               |-- huf_decompress.c.o
|   |   |   |   |   |               `-- zstd_decompress.c.o
|   |   |   |   |   |-- bitshuffle-avx2.c.o
|   |   |   |   |   |-- bitshuffle-generic.c.o
|   |   |   |   |   |-- bitshuffle-sse2.c.o
|   |   |   |   |   |-- blosc.c.o
|   |   |   |   |   |-- blosclz.c.o
|   |   |   |   |   |-- build.make
|   |   |   |   |   |-- cmake_clean.cmake
|   |   |   |   |   |-- cmake_clean_target.cmake
|   |   |   |   |   |-- depend.internal
|   |   |   |   |   |-- depend.make
|   |   |   |   |   |-- flags.make
|   |   |   |   |   |-- link.txt
|   |   |   |   |   |-- progress.make
|   |   |   |   |   |-- shuffle-avx2.c.o
|   |   |   |   |   |-- shuffle-generic.c.o
|   |   |   |   |   |-- shuffle-sse2.c.o
|   |   |   |   |   `-- shuffle.c.o
|   |   |   |   `-- progress.marks
|   |   |   |-- Makefile
|   |   |   |-- cmake_install.cmake
|   |   |   |-- config.h
|   |   |   `-- libblosc.a
|   |   `-- cmake_install.cmake
|   `-- pybind11
|       |-- CMakeFiles
|       |   |-- CMakeDirectoryInformation.cmake
|       |   `-- progress.marks
|       |-- Makefile
|       `-- cmake_install.cmake
|-- libapr.so -> libapr.so.1
|-- libapr.so.1 -> libapr.so.1.1.0
|-- libapr.so.1.1.0
`-- pyApr.so

35 directories, 119 files
root@9d00051e6719:/srv/LibAPR/build#
```

**Comment by krzysg** [comment]
Hi, do you see the same problem if you try import pyApr from your build directory (so cd to the place where pyApr.so is generated)
You should be able to import that:

```
~/APR/LibAPRPlayground/build (develop=)$ python2
Python 2.7.12 (default, Nov 12 2018, 14:36:49) 
[GCC 5.4.0 20160609] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import pyApr
>>> dir(pyApr)
['APRParameters', 'AprByte', 'AprFloat', 'AprShort', 'PixelDataByte', 'PixelDataFloat', 'PixelDataShort', '__doc__', '__file__', '__name__', '__package__', '__version__']
```

If you want to use pyApr from any other place setting PYTHONPATH should help:

```
~/APR/LibAPRPlayground (develop=)$ export PYTHONPATH=/home/gonciarz/APR/LibAPRPlayground/build:$PYTHONPATH
~/APR/LibAPRPlayground (develop=)$ python2
Python 2.7.12 (default, Nov 12 2018, 14:36:49) 
[GCC 5.4.0 20160609] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import pyApr
>>> dir(pyApr)
['APRParameters', 'AprByte', 'AprFloat', 'AprShort', 'PixelDataByte', 'PixelDataFloat', 'PixelDataShort', '__doc__', '__file__', '__name__', '__package__', '__version__']
```

One more important thing. By default wrappers are generated for python 2. So it can be also imported only by python 2. If you need it for python 3 then change  [this line](https://github.com/AdaptiveParticles/LibAPR/blob/c13429091a7c2481c27cef9f4f2a92a684ca5998/CMakeLists.txt#L323) in CMakeLists.txt and then build again.

**Comment by VolkerH** [comment]
Hi @krzysg, 

thanks for the quick reply. Both of your recommendations were spot-on.
When I `cd` to the build directory and invoke python from there I can import it. I don't know why it didn't work with setting the path using `sys.path.insert`. Maybe I mistyped the path. Will try again. 

Also, I was trying with Python 3, when I used python 2.7 I managed to import it. 

Thanks a lot, now I think I have everything in place to get started experimenting with pyApr.
